client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -535,7 +535,11 @@ export class CheckoutThankYou extends Component<
 		let wasDomainOnly = false;
 		let wasBulkDomainTransfer = false;
 
-		if ( this.isDataLoaded() && ! this.isGenericReceipt() ) {
+		if ( ! this.isDataLoaded() ) {
+			return <PlaceholderThankYou />;
+		}
```

Comment:
Earlier, the placeholder was split into different parts - in different components, we were checking something like `const classes = { 'is-placeholder': ! isDataLoaded };`.
https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/header.jsx#L490

I think that it doesn't make sense, at least now when we have the same structure of cngrats pages. And when we have just 1 encapsulated component - it's much easier to do maintenance in the future and the rendering flow is much more easy and straightforward.
So, I am introducing PlaceholderThankYou component and render it as a first thing when data isn't loaded. And, I will create a follow-up PR with cleaning up old placeholder parts from different modules.

@wongasy I need to know your opinion regarding it. If you are on board with my approach - I will merge this PR to the `trunk` and create a follow-up PR to clean up the old approach.

Reply:
Your approach makes sense.